### ISSUE TITLE
Highlighting property of hits object doesnt retain custom field name

### ISSUE THREAD

**Issue by guruprasathk** [sunspot/sunspot#431]
Using a custom name for fields does not work as expected when highlighting is requested.
text :content, :as => 'autocomplete_simple_token', :stored => :true do
     token
end

search.hits.each do |hit|
      logger.debug hit.highlights.to_json
        hit.highlights(:content).each do |highlight|

This doesnt work. When inspecting the hit.highlights object, the following is logged: [{"field_name":"autocomplete_simple","highlight":"@@@hl@@@bla@@@endhl@@@ bla"}]
Basically strips off the name after last underscore and forms the field name
Here the field name should ideally be "content" so that one is able to retrieve using hit.highlights(:content) instead of hit.highlight(:autocomplete_simple)
Version is 2.0.0

**Comment by TimWei** [comment]
We just facing the similar problem recently and patched the code that how highlights parsing its field name. 
Instead of split from last "_",  we compare indexed name from solr result with Sunspot Setup field factories cache to getting correct field names.
